wireapp/wire-android
# feat: migrate other users data from scala app [AR-2689]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2689" title="AR-2689" target="_blank"><img alt="Subtask" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10816?size=medium" />AR-2689</a>  Migrate other users data
  </td></table>
  <br />
 










----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

When migrating from old app version, we get all conversations and messages but not info about members of these conversations and creators of messages, sometimes the old app can still have some useful information about for instance removed members and we won't get info about them during sync anymore.

### Solutions

Implement `MigrateUsersUseCase` which adds missing users data, if such user has been fetched during the sync then just ignore. Updated `ScalaUserDAO` to include more data, `MigrationMapper` to add migration to kalium data objects and added the step to the `MigrationManager`.

Needs releases with:

- https://github.com/wireapp/kalium/pull/1129

### Testing

#### How to Test

Install and log in to the scala app and update to the AR version, the app should perform the migration successfully and the data of all users, even the ones that are not conversation members anymore (can be checked by looking at user names of messages sent by ones that are no longer in the group and not connected with you).

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
